src/AstEditor.tsx

```diff
@@ -1082,3 +1098,96 @@ function FunctionEditor({
     />
   );
 }
+
+function MathEditor(props: NodeEditorProps<BinaryNode>): JSX.Element {
+  const { theme, defaultProvider } = Context.useContainer();
+  const [text, setText] = useState(serializer(props.ast));
+  const [parsing, setParsing] = useState<ParsingState>({
+    inProgress: false,
+  });
+  const changeType = () => { props.onChange(nextAst(props.ast, defaultProvider)) };
+  const parts = flattenMathParts(props.ast, props.onChange);
+
+  function onChangeText(newText: string) {
+    let error: string | undefined = undefined;
+    setParsing({
+      inProgress: true,
+      error
+    });
+    try {
+      const newAst = jsonata(newText).ast() as AST;
+      if (!isMath(newAst)) {
+        throw new Error("that's not a math expressions");
+      }
+      props.onChange(newAst);
+    } catch (e) {
+      error = "Parsing Error: " + e.message;
+    } finally {
+     setParsing({
+       inProgress: false,
+       error
+     });
+     setText(newText);
+    }
+  }
+
+  return (
+    <theme.MathEditor 
+      text={text} 
+      children={parts}
+      textChange={onChangeText} 
+      parsing={parsing} 
+      changeType={changeType}
+      {...props}
+    />
+  );
+}
+
+function flattenMathParts(ast: AST, onChange: OnChange, collectedParts: MathPart[] = []): MathPart[] {
+  function onChangeOperator(newOperator: string) {
+    if (Object.keys(Consts.mathOperators).includes(newOperator)) {
+      onChange({ ...ast, value: newOperator } as BinaryNode);
+    } else {
+      throw new Error("Not a valid math operator");
+    }
+  }
+
+  if (ast.type == "binary") {
```

Comment:
I think that this is needed to avoid weirdness with an expression like `1 + 1 = true`

```suggestion
  if (ast.type == "binary" && Object.keys(Consts.mathOperators).includes(props.ast.value)) {
```

Reply:
Fixed